clinica/iotools/converters/ixi_to_bids/ixi_to_bids_utils.py

```diff
@@ -0,0 +1,209 @@
+import json
+import re
+import shutil
+from pathlib import Path
+from typing import List, Optional
+
+import nibabel as nib
+import pandas as pd
+from nilearn.image import concat_imgs
+
+from clinica.iotools.bids_utils import StudyName, bids_id_factory
+from clinica.utils.stream import cprint
+
+
+def get_subjects_list_from_data(data_directory: Path) -> List[str]:
+    return list(
+        dict.fromkeys(
+            re.match(r"IXI\d{3}", path.name).group(0)
+            for path in data_directory.rglob(pattern="IXI*.nii.gz")
+            if re.match(r"IXI\d{3}", path.name)
+        )
+    )
+
+
+def filter_subjects_list(
+    subjects_list: List[str], clinical_data: pd.DataFrame
+) -> List[str]:
+    return [
+        subject
+        for subject in subjects_list
+        if subject in clinical_data["ixi_id"].values
+    ]
+
+
+def define_participants(
+    data_directory: Path,
+    clinical_data: pd.DataFrame,
+    subjs_list_path: Optional[Path] = None,
+) -> List[str]:
+    """Define the actual list of participants based on the (provided) subjects list filtered
+    using clinical data"""
+    if subjs_list_path:
+        cprint("Loading a subjects list provided by the user...")
+        subjects_to_filter = subjs_list_path.read_text().splitlines()
+    else:
+        subjects_to_filter = get_subjects_list_from_data(data_directory)
+    return filter_subjects_list(subjects_to_filter, clinical_data)
+
+
+def read_ixi_clinical_data(clinical_data_path: Path) -> pd.DataFrame:
+    """Reads and formats IXI clinical data"""
+    # todo : what keys ?
```

Comment:
still relevant ?

Reply:
It is but the question is badly worded : 

I am currently using the clinical data as is to put inside the sessions.tsv. However the keys of this dataframe are probably not BIDS 'compliant'. I am unsure of :
- what keys should actually go in the sessions.tsv
- under what name 
- if I should retrieve the info that is masked by identifiers (ex : male/female rather than 1/2 for sex)

Reply:
Looking at the [specifications](https://bids-specification.readthedocs.io/en/stable/modality-agnostic-files.html#sessions-file), I'm not sure we need a `sessions.tsv` file since there is only one session per subject, right ?
If I understand correctly, it still possible to have it. The required columns and their names are indicated in the specs, and it is possible to add other columns if needed. 
However, metadata like sex or age should go in the [participants.tsv](https://bids-specification.readthedocs.io/en/stable/modality-agnostic-files.html#participants-file).

What columns do you have in the input dataframe ?

Reply:
Right, there is only one session per subject. That reminds me I forgot to write the participants.tsv file... 

I have sex, height, weight, ethnicity, marital situation, occupation, qualification, date of birth, age, study date